README.md
# Docker image with the pysmurf server for the SMuRF project

## Description

This docker image, named **pysmurf-server** contains pysmurf, firmware files (mcs and pyrogue's zip) and default configuration files (yml) for stable firmware versions.

It is based on the **pysmurf-server-base** docker image.

## Source code

The firmware files are checkout from the SLAC's github repositories https://github.com/slaclab/cryo-det.

The configuration files are checkout from the SMuRF's configuration repository https://github.com/slaclab/smurf_cfg.git.

If needed files are not available on any of the github repositories, then then are added to this repository under the [local_files](local_files) directory.

## Tag naming convention

The naming convention followed for tags is describe in [README.tagNamingConvention.md](README.tagNamingConvention.md).

## Building the image

When a tag is pushed to this github repository, a new Docker image is automatically built and push to its [Dockerhub repository](https://hub.docker.com/r/tidair/pysmurf-server) using travis.

The resulting docker image is tagged with the same git tag string (as returned by `git describe --tags --always`).

The image is built based on a user defined configuration, using the [definitions.sh](definitions.sh) file. In that file, the user can define:
- The `pysmurf-server-base` image version,
- The firmware image (mcs) file,
- The pyrogue definitnion (zip) file,
- The configuration (yml) file, and
- The startup arguments

For more information about how to defined these parameters see [README.buildDefinitions.md](README.buildDefinitions.md).

## How to get the container

To get the docker image, first you will need to install the docker engine in you host OS. Then you can pull a copy by running:

```
docker pull tidair/pysmurf-server:<TAG>
```

Where **TAG** represents the specific tagged version you want to use.

## Running the container

Each docker image defines which it's entry point. By default, the entry point calls the `start_server.sh` (which comes within the `pysmurf-server-base` image) script with some pre-defined arguments. You can however, overwrite any of the arguments and/or adding new one by passing them at the end of the docker run command.

You start the container with a command like this one:

```
docker run -ti --rm \
    -v <local_data_dir>:/data \
    tidair/pysmurf-server:<TAG> \
    <server_arguments>
```

Where:
- **local_data_dir**: is a local directory in the host CPU which contains the directories `smurf_data` where the data is going to be written to,
- **TAG**: is the tagged version of the container your want to run,
- **server_arguments**: additional and/or redefined server arguments.

RELEASES.md
# Released Version Description

You can also see the list of releases [here](https://github.com/slaclab/pysmurf-stable-docker/releases).

Version      | pysmurf-server-base version | MCS file                                                                        | pyrogue's zip file                                                                                            | configuration file                                | Server arguments
-------------|-----------------------------|---------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------|---------------------------------------------------|-----------------------------------
 v0.0.0      | v4.0.0-rc16                 | MicrowaveMuxBpEthGen2-0x00000020-20190916220655-mdewart-3854241 (local copy)    | MicrowaveMuxBpEthGen2-0x00000020-20190916220655-mdewart-3854241.zip (local copym incl. numpy imports fixes)   | defaults_lbonly_c03_bay0.yml (v0.0.3)             | --disable-bay1
 v0.0.1      | v4.0.0-rc16                 | MicrowaveMuxBpEthGen2-0x00000020-20190916220655-mdewart-3854241 (local copy)    | MicrowaveMuxBpEthGen2-0x00000020-20190916220655-mdewart-3854241.zip (local copym incl. numpy imports fixes)   | defaults_lbonly_c02_bay0.yml (v0.0.3)             | --disable-bay1
 v0.0.2      | v4.0.0-rc16                 | MicrowaveMuxBpEthGen2-0x00000020-20190916220655-mdewart-3854241 (local copy)    | MicrowaveMuxBpEthGen2-0x00000020-20190916220655-mdewart-3854241.zip (local copym incl. numpy imports fixes)   | 2019_06_04_Dual_Band_AMC_Config_LBx2.yml (v0.0.3) |
v1.0.0 | v4.0.0 | MicrowaveMuxBpEthGen2-0x00000020-20191203110805-mdewart-83947a3.mcs.gz (MicrowaveMuxBpEthGen2_v0.0.5) | rogue_MicrowaveMuxBpEthGen2_v0.0.5.zip (MicrowaveMuxBpEthGen2_v0.0.5) | Full repo (v1.0.0) | 
v2.0.0 | v4.1.0 | MicrowaveMuxBpEthGen2-0x00000031-20200707075937-ruckman-deac6ad.mcs.gz (MicrowaveMuxBpEthGen2_v0.3.0) | rogue_MicrowaveMuxBpEthGen2_v0.3.0.zip (MicrowaveMuxBpEthGen2_v0.3.0) | Full repo (v1.3.0) | 
v2.1.0 | v4.1.0 | MicrowaveMuxBpEthGen2-0x00000032-20200810095918-mdewart-e08c79a.mcs.gz (MicrowaveMuxBpEthGen2_v0.4.1) | rogue_MicrowaveMuxBpEthGen2_v0.4.1.zip (MicrowaveMuxBpEthGen2_v0.4.1) | Full repo (v1.3.0) | 
v3.0.0 | v4.2.1 | MicrowaveMuxBpEthGen2-0x00000100-20210106214208-mdewart-e8c5d46.mcs.gz (MicrowaveMuxBpEthGen2_v1.0.4) | rogue_MicrowaveMuxBpEthGen2_v1.0.4.zip (MicrowaveMuxBpEthGen2_v1.0.4) | Full repo (v1.4.0) | 
v4.0.0 | v4.3.0 | MicrowaveMuxBpEthGen2-0x00000100-20210106214208-mdewart-e8c5d46.mcs.gz (MicrowaveMuxBpEthGen2_v1.0.4) | rogue_MicrowaveMuxBpEthGen2_v1.0.4.zip (MicrowaveMuxBpEthGen2_v1.0.4) | Full repo (v1.4.0) | 
v5.0.0 | v5.0.0 | MicrowaveMuxBpEthGen2-0x00000100-20210106214208-mdewart-e8c5d46.mcs.gz (MicrowaveMuxBpEthGen2_v1.0.4) | rogue_MicrowaveMuxBpEthGen2_v1.0.4.zip (MicrowaveMuxBpEthGen2_v1.0.4) | Full repo (v1.4.0) | 


validate.sh
#!/usr/bin/env bash

# Validate definitions.sh

#############
# FUNCTIONS #
#############
# Check if a tag exists on a github public repository
# Arguments:
# - first: github public repository url,
# - second: tag name
check_if_public_tag_exist()
{
    local repo=$1
    local tag=$2
    git ls-remote --refs --tag ${repo} | grep -q refs/tags/${tag} > /dev/null
}

# Check if a asset file exist on a tag version on a github public repository
# Arguments:
# - first: github public repository url,
# - second: tag name,
# - third: asset file name
check_if_public_asset_exist()
{
    local repo=$1
    local tag=$2
    local file=$3
    curl --head --silent --fail ${repo}/releases/download/${tag}/${file} > /dev/null
}

# Check if a tag exists on a github pivate repository.
# It requires ssh key authentication.
# Arguments:
# - first: github private repository url,
# - second: tag name
check_if_private_tag_exist()
{
    # Need to insert the token in the url
    local repo=$(echo $1 | sed -e "s|https://github.com/|git@github.com:|g")
    local tag=$2
    git ls-remote --refs --tag ${repo} | grep -q refs/tags/${tag} > /dev/null
}

# Check if a asset file exist on a tag version on a github private repository.
# It requires the access token to be defined in $GITHUB_TOKEN.
# Arguments:
# - first: github private repository url,
# - second: tag name,
# - third: asset file name
check_if_private_asset_exist()
{
    local repo=$(echo $1 | sed -e "s|https://github.com|https://api.github.com/repos|g")
    local tag=$2
    local file=$3

    # Search the asset ID in the specified release
    local r=$(curl --silent --header "Authorization: token ${GITHUB_TOKEN}" "${repo}/releases/tags/${tag}")
    eval $(echo "${r}" | grep -C3 "name.:.\+${file}" | grep -w id | tr : = | tr -cd '[[:alnum:]]=')

    # return is the asset tag was found
    [ "${id}" ]
}

# Download the asset file on a tagged version on a github private repository.
# It requires the access token to be defined in $GITHUB_TOKEN.
# Arguments:
# - first: github private repository url,
# - second: tag name,
# - third: asset file name
get_private_asset()
{
    local repo=$(echo $1 | sed -e "s|https://github.com|https://api.github.com/repos|g")
    local tag=$2
    local file=$3

    # Check if the asset exist, and get it's ID
    check_if_private_asset_exist ${repo} ${tag} ${file} || exit 1

    echo "Downloading ${file}..."

    # Try to download the asset
    curl --fail --location --remote-header-name --remote-name --progress-bar \
         --header "Authorization: token ${GITHUB_TOKEN}" \
         --header "Accept: application/octet-stream" \
         "${repo}/releases/assets/${id}"
}

# Check if file exist on a tag version on a github repository
# Arguments:
# - first: github repository url,
# - second: tag name,
# - third: file name (must be a full path on that repository)
check_if_file_exist()
{
    local repo=$1
    local tag=$2
    local file=$3
    curl --head --silent --fail ${repo}/blob/${tag}/${file} > /dev/null
}

# Exit with an error message, and with return code = 1
exit_on_error()
{
    echo
    echo "Validation failed! The definitions.sh file is incorrect!"
    echo
    exit 1
}

# Exist with a success message and with return core = 0
exit_on_success()
{
    echo
    echo "Success! The definitions.sh file is correct!"
    echo
}

#############
# MAIN BODY #
#############

# Load the user definitions
. definitions.sh

# Validate base docker image
echo "==========================================="
echo "Validating pysmurf-server-base version..."
echo "==========================================="

printf "Checking is tagged release exist...       "
check_if_public_tag_exist ${pysmurf_repo} ${pysmurf_server_base_version}
if [ $? != 0 ]; then
    echo "Failed!"
    echo
    echo "Release \"${pysmurf_server_base_version}\" does not exist in repository \"${pysmurf_repo}\"!"
    exit_on_error
fi

echo "Release exist!"

# At this point the definition of the MCS file is correct,
# We will use a local file.
echo "Done!"
echo
echo "A correct pysmurf-server-base version was defined. This version will be used"
echo "  tidair/pysmurf-server-base:${pysmurf_server_base_version}"
echo "==========================================="
echo

# Validate MCS file selection
echo "==========================================="
echo "Validating MCS file selection..."
echo "==========================================="

printf "Checking if a tag was defined...          "
if [ -z ${mcs_repo_tag} ]; then
    echo "A tag was not defined."
    printf "Looking for a local file...               "

    local_mcs_files=$(find local_files -name *.mcs -o -name *.mcs.gz)
    num_local_mcs_files=$(echo -n "${local_mcs_files}" | grep -c '^')

    if [ ${num_local_mcs_files} == 0 ]; then
        echo "Failed!"
        echo
        echo "No local MCS file was found!"
        exit_on_error
    elif [ ${num_local_mcs_files} != 1 ]; then
        echo "Failed!"
        echo
        echo "More than one local MCS file was found!"
        exit_on_error
    else
        mcs_file_name=$(basename ${local_mcs_files})

        echo "File \"${mcs_file_name}\" was found!"

        # At this point the definition of the MCS file is correct,
        # We will use a local file.
        echo "Done!"
        echo
        echo "A correct MCS was defined. This local file will be used:"
        echo "  ${mcs_file_name}"
        echo "==========================================="
        echo

        # Variables passed to the build.sh script.
        mcs_use_local=1
    fi
else
    echo "Tag \"${mcs_repo_tag}\" was defined."

    printf "Checking is file name was defined...      "
    if [ -z ${mcs_file_name} ]; then
        echo "Failed!"
        echo
        echo "File name was not defined!"
        exit_on_error
    else
        echo "File name \"${mcs_file_name}\" was defined."
    fi

    printf "Checking is tagged release exist...       "
    check_if_private_tag_exist ${mcs_repo} ${mcs_repo_tag}
    if [ $? == 0 ]; then
        echo "Release exist!"
        printf "Checking if file exist on that release... "
        check_if_private_asset_exist ${mcs_repo} ${mcs_repo_tag} ${mcs_file_name}
        if [ $? == 0 ]; then
            echo "File exist!"

            # At this point the definition of the MCS file is correct,
            # We will use a file from the git repository.
            echo "Done!"
            echo
            echo "A correct MCS was defined. This remote file will be used:"
            echo "  ${mcs_repo}/releases/download/${mcs_repo_tag}/${mcs_file_name}"
            echo "==========================================="
            echo
        else
            echo "Failed!"
            echo
            echo "File \"${mcs_file_name}\" does not exist in release \"${mcs_repo_tag}\", repository \"${mcs_repo}\"!"
            exit_on_error
        fi

    else
        echo "Failed!"
        echo
        echo "Release \"${mcs_repo_tag}\" does not exist in repository \"${mcs_repo}\"!"
        exit_on_error
    fi
fi

# Validate ZIP file selection
echo "==========================================="
echo "Validating ZIP file selection..."
echo "==========================================="

printf "Checking if a tag was defined...          "
if [ -z ${zip_repo_tag} ]; then
    echo "A tag was not defined."
    printf "Looking for a local file...               "

    local_zip_files=$(find local_files -name *.zip)
    num_local_zip_files=$(echo -n "${local_zip_files}" | grep -c '^')

    if [ ${num_local_zip_files} == 0 ]; then
        echo "Failed!"
        echo
        echo "No local ZIP file was found!"
        exit_on_error
    elif [ ${num_local_zip_files} != 1 ]; then
        echo "Failed!"
        echo
        echo "More than one local ZIP file was found!"
        exit_on_error
    else
        zip_file_name=$(basename ${local_zip_files})

        echo "File \"${zip_file_name}\" was found!"

        # At this point the definition of the ZIP file is correct,
        # We will use a local file.
        echo "Done!"
        echo
        echo "A correct ZIP was defined. This local file will be used:"
        echo "  ${zip_file_name}"
        echo "==========================================="
        echo

        # Variables passed to the build.sh script.
        zip_use_local=1
    fi
else
    echo "Tag \"${zip_repo_tag}\" was defined."

    printf "Checking is file name was defined...      "
    if [ -z ${zip_file_name} ]; then
        echo "Failed!"
        echo
        echo "File name was not defined!"
        exit_on_error
    else
        echo "File name \"${zip_file_name}\" was defined."
    fi

    printf "Checking is tagged release exist...       "
    check_if_private_tag_exist ${zip_repo} ${zip_repo_tag}
    if [ $? == 0 ]; then
        echo "Release exist!"
        printf "Checking if file exist on that release... "
        check_if_private_asset_exist ${zip_repo} ${zip_repo_tag} ${zip_file_name}
        if [ $? == 0 ]; then
            echo "File exist!"

            # At this point the definition of the ZIP file is correct,
            # We will use a file from the git repository.
            echo "Done!"
            echo
            echo "A correct ZIP was defined. This remote file will be used:"
            echo "  ${zip_repo}/releases/download/${zip_repo_tag}/${zip_file_name}"
            echo "==========================================="
            echo
        else
            echo "Failed!"
            echo
            echo "File \"${zip_file_name}\" does not exist in release \"${zip_repo_tag}\", repository \"${zip_repo}\"!"
            exit_on_error
        fi

    else
        echo "Failed!"
        echo
        echo "Release \"${zip_repo_tag}\" does not exist in repository \"${zip_repo}\"!"
        exit_on_error
    fi
fi

# Validate ZIP file selection
echo "==========================================="
echo "Validating YML file selection..."
echo "==========================================="

printf "Checking if a tag was defined...          "
if [ -z ${yml_repo_tag} ]; then
    echo "A tag was not defined."
    printf "Looking for a local file...               "

    local_yml_files=$(find local_files -name *.yml)
    num_local_yml_files=$(echo -n "${local_yml_files}" | grep -c '^')

    if [ ${num_local_yml_files} == 0 ]; then
        echo "Failed!"
        echo
        echo "No local YML file was found!"
        exit_on_error
    elif [ ${num_local_yml_files} != 1 ]; then
        echo "Failed!"
        echo
        echo "More than one local YML file was found!"
        exit_on_error
    else
        yml_file_name=$(basename ${local_yml_files})

        echo "File \"${yml_file_name}\" was found!"

        # At this point the definition of the ZIP file is correct,
        # We will use a local file.
        echo "Done!"
        echo
        echo "A correct YML was defined. This local file will be used:"
        echo "  ${yml_file_name}"
        echo "==========================================="
        echo

        # Variables passed to the build.sh script.
        yml_use_local=1
    fi
else
    echo "Tag \"${yml_repo_tag}\" was defined."
fi

printf "Checking is tagged release exist...       "
check_if_public_tag_exist ${yml_repo} ${yml_repo_tag}
if [ $? == 0 ]; then
    echo "Release exist!"
else
    echo "Failed!"
    echo
    echo "Release \"${yml_repo_tag}\" does not exist in repository \"${yml_repo}\"!"
    exit_on_error
fi

printf "Checking is file name was defined...      "
if [ -z ${yml_file_name} ]; then
    echo "An specific file was not defined."

    # At this point the definition of the YML file is correct.
    # We are not going to use an specific file name from the repository.
    echo "Done!"
    echo
    echo "A correct YML repository and tag were defined. An specific YML will not be used."
    echo "We will use the repository ${yml_repo}, tagged version ${yml_repo_tag}"
    echo "==========================================="
    echo
else
    echo "File name \"${yml_file_name}\" was defined."

    printf "Checking if file exist on that release... "
    check_if_file_exist ${yml_repo} ${yml_repo_tag} "defaults/${yml_file_name}"
    if [ $? == 0 ]; then
        echo "File exist!"

        # At this point the definition of the YML file is correct,
        # We will use a file from the git repository.
        echo "Done!"
        echo
        echo "A correct YML repository, tag, and specific file were defined."
        echo "This remote file will be used:"
        echo "  ${yml_repo}/blob/${yml_repo_tag}/defaults/${yml_file_name}"
        echo "==========================================="
        echo
    else
        echo "Failed!"
        echo
        echo "File \"${yml_file_name}\" does not exist in release \"${yml_repo_tag}\", repository \"${yml_repo}\"!"
        exit_on_error
    fi
fi

# At this point all definition where correct
exit_on_success


scripts/build.sh
#!/usr/bin/env bash

# Load the user definitions
. definitions.sh

# Call the validation script
. validate.sh

# Check if the required environmental variables are defined.
if [ -z ${DOCKERHUB_ORG_NAME+x} ]; then
    echo "ERROR: DOCKERHUB_ORG_NAME environmental variable not defined!"
    exit 1
fi

if [ -z ${DOCKERHUB_REPO+x} ]; then
    echo "ERROR: DOCKERHUB_REPO environmental variable not defined!"
    exit 1
fi

if [ -z ${REPO_SLUG+x} ]; then
    echo "ERROR: REPO_SLUG environmental variable not defined!"
    exit 1
fi

# Get the git tag, which will be used to tag the docker image
tag=`git describe --tags --always`

# Get mcs file
if [ -z ${mcs_use_local+x} ]; then
    echo "Getting mcs file from ${mcs_repo}"

    # Get the mcs file assent
    (cd local_files && get_private_asset ${mcs_repo} ${mcs_repo_tag} ${mcs_file_name}) || exit 1
else
    echo "Using local mcs file..."
fi

# Get zip file
if [ -z ${zip_use_local+x} ]; then
    echo "Getting zip file from ${zip_repo}"

    # Get the zip file asset
    (cd local_files && get_private_asset ${zip_repo} ${zip_repo_tag} ${zip_file_name}) || exit 1
else
    echo "Using local zip file..."
fi

# Get yml file
if [ -z ${yml_use_local+x} ]; then
    echo "Getting yml file from ${yml_repo}"

    if [ -z ${yml_file_name} ]; then
        # If an specific file was not define, clone the whole repository
        git -C local_files clone ${yml_repo} -b ${yml_repo_tag} || exit 1
    else
        # If an specific file was defined, copy it
        git clone ${yml_repo} -b ${yml_repo_tag} yml_repo || exit 1
        mv yml_repo/defaults/${yml_file_name} local_files || exit 1
        rm -rf yml_repo

        # Additionally, when an specific YML file is defined, add the '--disable-hw-detect'
        # option to the list of server arguments as well as the specified YML file using the
        # '-d' option.
        server_args+=" --disable-hw-detect -d /tmp/fw/${yml_file_name}"
    fi

else
    echo "Using local yml file..."
fi

# Remove any white spaces and the beginning or end of the server argument string.
server_args=$(echo ${server_args} | sed 's/^\s//g'| sed 's/\s$//g')

# Then divide it into a list of quoted substring, divided by comas.
# This is the format that the Dockerfile uses
if [ -z "${server_args}" ]; then
    # If the server argument is empty, then the list will be empty
    server_args_list=""
else
    # Otherwise, generate the list.
    server_args_list=$(echo ,\"${server_args}\" | sed 's/\s/","/g')
fi

# Generate the Dockerfile from the template
cat Dockerfile.template \
        | sed "s|%%SERVER_ARGS%%|"${server_args_list}"|g" \
        | sed "s|%%SERVER_ARGS_ENV%%|${server_args}|g" \
        | sed "s|%%PYSMURF_SERVER_BASE_VERSION%%|${pysmurf_server_base_version}|g" \
        > Dockerfile

# Build the docker image and push it to Dockerhub
docker build -t ${DOCKERHUB_ORG_NAME}/${DOCKERHUB_REPO} . || exit 1
docker tag ${DOCKERHUB_ORG_NAME}/${DOCKERHUB_REPO} ${DOCKERHUB_ORG_NAME}/${DOCKERHUB_REPO}:${tag} || exit 1
docker push ${DOCKERHUB_ORG_NAME}/${DOCKERHUB_REPO}:${tag} || exit 1

echo "Docker image '${DOCKERHUB_ORG_NAME}/${DOCKERHUB_REPO}:${tag}' pushed"

# Update the release information
. scripts/generate_release_info.sh

scripts/generate_release_info.sh
#!/usr/bin/env bash

# Release information:

## Release name
release_name="${DOCKERHUB_ORG_NAME}/${DOCKERHUB_REPO}:${tag}"

## Release description and New entry on the RELEASE.md file
release_description="- **pysmurf-server-base:** [${pysmurf_server_base_version}](${pysmurf_repo}/releases/tag/${pysmurf_server_base_version})"
release_new_row="${tag} | ${pysmurf_server_base_version} | "

release_description+="
- **MCS file:** "
release_new_row+="${mcs_file_name} "
if [ -z ${mcs_use_local+x} ]; then
    release_description+="[${mcs_file_name}](${mcs_repo}/releases/download/${mcs_repo_tag}/${mcs_file_name}) (tag [${mcs_repo_tag}](${mcs_repo}/releases/tag/${mcs_repo_tag}))"
    release_new_row+="(${mcs_repo_tag}) | "
else
    release_description+="Local copy [${mcs_file_name}](https://github.com/${REPO_SLUG}/blob/${tag}/local_files/${mcs_file_name})"
    release_new_row+="(local copy) | "
fi

release_description+="
- **ZIP file:** "
release_new_row+="${zip_file_name} "
if [ -z ${zip_use_local+x} ]; then
    release_description+="[${zip_file_name}](${zip_repo}/releases/download/${zip_repo_tag}/${zip_file_name}) (tag [${zip_repo_tag}](${zip_repo}/releases/tag/${zip_repo_tag}))"
    release_new_row+="(${zip_repo_tag}) | "
else
    release_description+="Local copy [${zip_file_name}](https://github.com/${REPO_SLUG}/blob/${tag}/local_files/${zip_file_name})"
    release_new_row+="(local copy) | "
fi

release_description+="
- **YML file:** "
if [ -z ${yml_use_local+x} ]; then
    if [ -z ${yml_file_name} ]; then
        release_description+="Full repo ${yml_repo}"
        release_new_row+="Full repo (${yml_repo_tag}) | "
    else
        release_description+="[${yml_file_name}](${yml_repo}/blob/${yml_repo_tag}/defaults/${yml_file_name})"
        release_new_row+="${yml_file_name} (${yml_repo_tag}) | "
    fi
    release_description+=" (tag [${yml_repo_tag}](${yml_repo}/releases/tag/${yml_repo_tag}))"
else
    release_description+="Local copy [${yml_file_name}](https://github.com/${REPO_SLUG}/blob/${tag}/local_files/${yml_file_name})"
    release_new_row+="(local copy) | "
fi

release_description+="
- **Server arguments:** "
if ! [ -z "${server_args}" ]; then
    release_description+="\\\`${server_args}\\\`"
fi
release_new_row+="${server_args}"


## Write results to an output file
cat << EOF > vars.env
release_name="${release_name}"
release_description="${release_description}"
release_new_row="${release_new_row}"
EOF


definitions.sh
#!/usr/bin/env bash

# Define the pysmurf-server-base version
# ======================================
# - Set the 'pysmurf_server_base_version' variable to the
#   pysmurf-server-base docker image version you want to use
#   as a base.
# - Normally, you should need to change the repository URL.
pysmurf_repo=https://github.com/slaclab/pysmurf
pysmurf_server_base_version=v5.0.0

# Define the MCS file location:
# =============================
# - Set the repository, tag, and name of the MCS file you want to
#   use. It will be downloaded from github repository's assets.
# - You can use a local file instead. In this case you must leave
#   the 'mcs_repo_tag' variable empty, and provide a local copy of
#   the MCS file in the 'local_files' directory.
# - If more that one local file is provided, the building process will
#   failed, so make sure only one MCS file exist in that directory.
# - Valid MCS file have extension '.mcs' or '.mcs.gz'.
# - If 'mcs_repo_tag' is defined, local files will be ignored.
# - Normally, you should need to change the repository URL.
mcs_repo=https://github.com/slaclab/cryo-det
mcs_repo_tag=MicrowaveMuxBpEthGen2_v1.0.4
mcs_file_name=MicrowaveMuxBpEthGen2-0x00000100-20210106214208-mdewart-e8c5d46.mcs.gz

# Define the ZIP file location:
# =============================
# - Set the repository, tag, and name of the ZIP file you want to
#   use. It will be downloaded from the github repository's assets.
# - You can use a local file instead. In this case you must leave
#   the 'zip_repo_tag' variable empty, and provide a local copy of
#   the ZIP file in the 'local_files' directory.
# - If more that one local file is provided, the building process will
#   failed, so make sure only one ZIP file exist in that directory.
# - Valid ZIP file have extension '.zip'.
# - If 'zip_repo_tag' is defined, local files will be ignored.
# - This file usually comes from the same repository and the same tag
#   as the MCS file.
# - Normally, you should need to change the repository URL.
zip_repo=${mcs_repo}
zip_repo_tag=${mcs_repo_tag}
zip_file_name=rogue_${mcs_repo_tag}.zip

# Define the YML file location:
# =============================
# - Set the repository and tag you want to use. The repository
#   will be cloned from github.
# - You can also defined an specific YAML file name, If so, it
#   will be added to the server startup argument list.
# - You can use a local file instead. In this case you must leave
#   the 'yml_repo_tag' variable empty, and provide a local copy of
#   the YML file in the 'local_files' directory.
# - If more that one local file is provided, the building process will
#   failed, so make sure only one YML file exist in that directory.
# - Valid YML file have extension '.yml'.
# - If 'yml_repo_tag' is defined, local files will be ignored.
# - Normally, you should need to change the repository URL.
yml_repo=https://github.com/slaclab/smurf_cfg
yml_repo_tag=v1.4.0
yml_file_name=

# Define server startup arguments
# ===============================
# - Add here a string with all the wanted startup arguments.
#   Note that if you defined an specific YML file, then the arguments
#   '--disable-hw-detect' as well as '-d' pointing to that file will be
#   automatically added. So, you don't need to added then here.
server_args=""
